hack/generate-operator-bundle.py

```diff
@@ -4,115 +4,199 @@
+    with open(HIVEADMISSION_ROLE, 'r') as stream:
+        hiveadmission_role = yaml.load(stream, Loader=yaml.SafeLoader)
+        csv['spec']['install']['spec']['clusterPermissions'].append(
+            {
+                'rules': hiveadmission_role['rules'],
+                'serviceAccountName': 'hiveadmission',
+            })
+
+    # Add our hiveapi role to the CSV:
+    with open(APISERVER_ROLE, 'r') as stream:
+        hiveapi_role = yaml.load(stream, Loader=yaml.SafeLoader)
+        csv['spec']['install']['spec']['clusterPermissions'].append(
+            {
+                'rules': hiveapi_role['rules'],
+                'serviceAccountName': 'hiveapi-sa',
+            })
+
+    # Add our deployment spec for the hive operator:
+    with open(DEPLOYMENT_SPEC, 'r') as stream:
+        operator_components = []
+        operator = yaml.load_all(stream, Loader=yaml.SafeLoader)
+        for doc in operator:
+            operator_components.append(doc)
+        operator_deployment = operator_components[1]
+        csv['spec']['install']['spec']['deployments'][0]['spec'] = operator_deployment['spec']
+
+    # Update the versions to include git hash:
+    csv['metadata']['name'] = "hive-operator.v%s" % version
+    csv['spec']['version'] = version
+    csv['spec']['replaces'] = "hive-operator.v%s" % prev_version
+
+    # Update the deployment to use the defined image:
+    image_ref = "%s:v%s" % (hive_image, version)
+    csv['spec']['install']['spec']['deployments'][0]['spec']['template']['spec']['containers'][0]['image'] = image_ref
+    csv['metadata']['annotations']['containerImage'] = image_ref
+
+    # Set the CSV createdAt annotation:
+    now = datetime.datetime.now()
+    csv['metadata']['annotations']['createdAt'] = now.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # Write the CSV to disk:
+    csv_filename = "hive-operator.v%s.clusterserviceversion.yaml" % version
+    csv_file = os.path.join(version_dir, csv_filename)
+    with open(csv_file, 'w') as outfile:
+        yaml.dump(csv, outfile, default_flow_style=False)
+    print("Wrote ClusterServiceVersion: %s" % csv_file)
+
+
+def update_package(version):
+    PACKAGE_FILE = 'bundle/hive.package.yaml'
+
+    with open(PACKAGE_FILE, 'r') as stream:
+        package = yaml.load(stream, Loader=yaml.SafeLoader)
+
+    for channel in package['channels']:
+        channel['currentCSV'] = "hive-operator.v%s" % version
+
+    with open(PACKAGE_FILE, 'w') as outfile:
+        yaml.dump(package, outfile, default_flow_style=False)
+    print("Updated package: %s" % PACKAGE_FILE)
+
+
+parser = argparse.ArgumentParser(
+    prog='OpenShift Hive Operator CSV generator',
+    description='created CSV and related artifacts for operatorhub.io and OpenShift Dedicated.',
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+)
+
+subparsers = parser.add_subparsers()
+
+osd_parser = subparsers.add_parser('generate-osd', help='Generate CSV for OpenShift Dedicated deployment.')
```

Comment:
Stutters with the script name, maybe just generate-operator-bundle.py osd or operatorhub

Reply:
great point. done.